gateway/action-handler.go

```diff
@@ -59,26 +60,76 @@ func (handler *actionHandler) Handler() gin.HandlerFunc {
 	logger := handler.context.Logger()
 
 	return func(ctx *gin.Context) {
-		handler.sendReponse(logger, <-handler.context.Call(handler.action, paramsFromRequest(ctx.Request, logger)), ctx.Writer)
+		logRequestFormatType, logRequestFormatTypeExists := handler.settings["logRequest"].(nucleo.LogLevelType)
+		if logRequestFormatTypeExists {
+			logRequestLogger := getLogger(logRequestFormatType, logger)
+			logRequestLogger("Call '", handler.action, "' action")
+		}
+
+		params := paramsFromRequest(ctx.Request, logger)
+
+		logRequestParamsFormatType, logRequestParamsFormatTypeExists := handler.settings["logRequestParams"].(nucleo.LogLevelType)
+		if logRequestParamsFormatTypeExists {
+			logRequestParamsLogger := getLogger(logRequestParamsFormatType, logger)
+			logRequestParamsLogger("Params: ", params)
+		}
+
+		callActionResponse := <-handler.context.Call(handler.action, params)
+
+		logResponseDataFormatType, logResponseDataFormatTypeExists := handler.settings["logResponseData"].(nucleo.LogLevelType)
+		if logResponseDataFormatTypeExists {
+			logResponseDataLogger := getLogger(logResponseDataFormatType, logger)
+			logResponseDataLogger("Data: ", callActionResponse)
+		}
+
+		handler.sendReponse(logger, callActionResponse, ctx)
 	}
 }
 
 var succesStatusCode = 200
 var errorStatusCode = 500
 
+func (handler *actionHandler) responesErrorHandler(ginContext *gin.Context, result nucleo.Payload) {
+	logger := handler.context.Logger()
+
+	ginContext.Writer.WriteHeader(errorStatusCode)
+
+	log4XXResponses, log4XXResponsesExists := handler.settings["log4XXResponses"].(nucleo.LogLevelType)
+	if log4XXResponsesExists {
+		log4XXResponsesLogger := getLogger(log4XXResponses, logger)
+		log4XXResponsesLogger("Gateway  Request error! - action: ", handler.action, " error ", result.Error())
+	}
+
+	// return response regardless.
+	json := jsonSerializer.PayloadToBytes(payload.Empty().Add("error", result.Error().Error()))
+	ginContext.Writer.Write(json)
+
+	onError, onErrorExists := handler.settings["onError"].(func(string))
+
+	// if user has onError middleware configured, they will be ablle to override it.
+	if onErrorExists {
+		// FIXME: parsing issue in nucleo-go
+		// onError(ginContext, result)
+		onError("string")
+	}
```

Comment:
![Screenshot 2023-11-12 at 00 03 23](https://github.com/Bendomey/awesome-nucleo/assets/41929243/9258a8ec-917d-4a89-bbf4-2cd64da3bd26)

Error that I gotta fix later. Just here for posterity